blog/_posts/2025-05-13-health-probe-improvements.md

```diff
@@ -0,0 +1,74 @@
+- kube‑proxy drift, network‑policy drops, or a single scaled‑to‑zero deployment could quietly take nodes out of rotation.  
+- Large clusters burned SNAT ports and SLB body size on thousands of probes.  
+- Because probes hit `nodePort`, you couldn’t set `allocateLoadBalancerNodePorts: false` for a performance win.
+
+## What we’re changing
+
+We’re flipping the probe around:
+
+- **New default** for `externalTrafficPolicy: Cluster` – SLB sends a single shared **HTTP** probe to `http://<nodeIP>:10256/healthz`.  
+  - A lightweight AKS sidecar handles PROXY‑protocol edge cases so Private Link Service scenarios keep working.
+  - That endpoint is kube‑proxy’s own health check.  
+
+- **Unchanged** – Services using `externalTrafficPolicy: Local` keep their per‑Service `healthCheckNodePort` behavior.
+
+### Why this is better
+
+| Challenge                      | Old behavior                 | New behavior                   |
+|--------------------------------|------------------------------|--------------------------------|
+| Probe reflects **node** health | ❌ Indirect (app‑level)       | ✅ Direct (kube‑proxy)          |
+| Duplicate configuration        | ❌ Yes                        | ✅ No                           |
+| Diagnose kube‑proxy drift      | ❌ Hard                       | ✅ SLB shows DIP failures       |
+| Node drain / autoscaler grace  | ❌ Black‑holes possible       | ✅ “Admin‑down” future‑proofing |
+| SLB probe bloat                | ❌ Every Service × every node | ✅ One probe per node           |
+| Disable extra NodePorts        | ❌ Blocked                    | ✅ Unblocked                    |
+
+## How it works under the hood
+
+1. **Shared probe** – Cloud‑provider config switches to *node health check* mode.  
+2. **Sidecar** – AKS deploys `node-health-proxy` alongside kube‑proxy:  
+   - Listens on a new port (defaults to `29999`).  
+   - Strips or parses the PROXY PDU so kube‑proxy doesn’t choke.  
+   - Forwards `/healthz` to localhost `10256`.
+
+## What this means for you
+
+- **Cleaner upgrades** – You no longer have to remember arcane SLB annotations when you migrate an ingress controller.  
+- **Observability you can trust** – A red DIP now really means “this node is unhealthy,” not “one of my pods failed HTTP on `/ready`.”  
+- **Fewer moving parts** – Turn off unneeded NodePorts, reduce SNAT churn, and shrink SLB config size.  
+
+## Getting started
+
+For more information on how to use the new health probe, check out the [AKS documentation](https://learn.microsoft.com/azure/aks/load-balancer-standard).
+
+## Summary
+
+By pointing SLB’s probe at the node instead of every app, AKS:
+
+- Removes a whole class of misconfigured probes.
+- Surfaces node health faster and more accurately.
+- Cuts probe traffic and SNAT pressure.
+- Opens the door to cleaner node drains and autoscaler events.
+
+Try it out today and let us know how it works in your environment!
```

Comment:
Here should link to AKS doc for the detailed guidance

Reply:
Do we need to include our open source doc link as well? Same for multi-slb doc.